### ISSUE TITLE
Expected behaviour of RETAIN_DETACH in child controllers

### ISSUE THREAD

**Issue by stephanschuster** [bluelinelabs/Conductor#277]
Given:
- ParentController with RetainViewMode.RELEASE_DETACH
- ChildController with RetainViewMode.RETAIN_DETACH

What happens:
- If PC is detached, CC.onDestroyView() is called even though it has RETAIN_DETACH set.
- This is because PC.removeHost() calls CC.detach() with "forceViewRefRemoval=true".

Questions:
- Is this on purpose?
- Is this correct because the view of PC has been destroyed and therefore the container of the CC?
- Wouldn't it be possible for conductor to NOT call CC.onDestroyView() and CC.onCreateView() in such cases (because of RETAIN_DETACH) but reuse the initialy created view of CC?

Otherwise RETAIN_DETACH on child controllers only works if all parent controllers also have RETAIN_DETACH set which might not always be desired (e.g. if there are multiple child controllers, some with RETAIN_DETACH and some with RELEASE_DETACH; or because the parent controllers just should not be aware of a child controllers view mode settings).

Would be good to have some clarification on that.
Thank you.

**Comment by EricKuck** [comment]
I think you're right and this is a great change to make. It does, however, require a fair bit of refactoring, so it may not happen right away. Thanks for the issue!

**Comment by sydneyagcaoili** [comment]
Up on this one

**Comment by StephanSchuster** [comment]
@EricKuck Are there any news/plans/progress on this topic? Would be great and super useful.

**Comment by StephanSchuster** [comment]
This issue is quite old meanwhile but still valid. Just as a reminder ...
